### PR TITLE
build: enable compiling on ARM

### PR DESCRIPTION
**Pull request by joergsteffens**
While trying to compile and test (ctest) Bareos on ARM (using qemu), I had to do 2 small changes to make it work.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General
- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Check backport line
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Separate commit for CHANGELOG.md ("update CHANGELOG.md"). The PR number is correct.

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems

### CODE REVIEW

File: core/src/droplet/libdroplet/src/backend/posix/backend.c [hunk 1]
```diff
@@ -144,7 +144,7 @@ dpl_status_t dpl_posix_head_raw(dpl_ctx_t* ctx,
     goto end;
   }
 
-  snprintf(buf, sizeof(buf), "%ld", st.st_nlink);
+  snprintf(buf, sizeof(buf), "%lu", (unsigned long)st.st_nlink);
```
Comment (pstorz) [inline core/src/droplet/libdroplet/src/backend/posix/backend.c:147]:
```suggestion
  snprintf(buf, sizeof(buf), "%lu", static_cast<unsigned long>(st.st_nlink));
```

File: core/src/droplet/libdroplet/src/backend/posix/backend.c [hunk 2]
```diff
@@ -179,7 +179,7 @@ dpl_status_t dpl_posix_head_raw(dpl_ctx_t* ctx,
     goto end;
   }
 
-  snprintf(buf, sizeof(buf), "%lu", st.st_blksize);
+  snprintf(buf, sizeof(buf), "%ld", (long int)st.st_blksize);
```
Comment (pstorz) [inline core/src/droplet/libdroplet/src/backend/posix/backend.c:182]:
```suggestion
  snprintf(buf, sizeof(buf), "%ld", static_cast<long int>(st.st_blksize));
```

**Review by pstorz — CHANGES_REQUESTED** [review]
Better use static_cast instead of c casts, see https://google.github.io/styleguide/cppguide.html#Casting

**Comment by joergsteffens** [comment]
Applied suggestions, appiled bareos-check-sources, rebased and force-pushed (because it is a trivial PR).

**Pull request merged**